### DISCUSSION TITLE
should GDL abort when started inside a not existing directory or continue as in IDL?

### DISCUSSION THREAD

**Discussion by brandy125** [gnudatalanguage/gdl#1711] [Q&A]
Today I noticed that GDL aborts when started inside a directory that does not exist any more:

```
bash$ mkdir testdir
bash$ cd testdir
bash$ gdl
  GDL - GNU Data Language, Version 1.0.4 git
GDL> exit
bash$ rmdir ../testdir
bash$ gdl

*** PLPLOT ERROR, IMMEDIATE EXIT ***
get_tmpdir_list: getcwd error
Program aborted
bash$ 
```

IDL continues in this case. 
What do you think?

**Comment by GillesDuvert** [comment]
a plplot error. Probably easy to correct by plplot maintainers.
For the record, IDL says:
```
% Unable to get working directory.
  No such file or directory
```

**Comment by alaingdl** [comment]
I suggest we add these lines just after
` int main(int argc, char *argv[]) ` in gdl.c : 

```
  char wd[PATH_MAX];
  char *wdp;
  wdp = getcwd(wd, PATH_MAX);
  if (wdp == NULL && errno == ENOENT) {
    cout << "Current dir was removed !" <<endl;
    exit(0);
  }
```

**Reply by GillesDuvert** [reply]
@alaingdl  the message should be, for compatibility "% Unable to get working directory.".
But IDL continues after the message, not exit(0). And one is thus obliged to check wdp at all places the current directory is needed, at least on exit of GDL.
